src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -179,6 +181,30 @@ export function registerActiveInstanceAction(
 	});
 }
 
+/**
+ * A wrapper around {@link registerTerminalAction} that ensures an active instance exists and
+ * provides it to the run function.
+ */
+export function registerActiveTerminalAction(
+	options: IAction2Options & { run: (activeTerminal: XtermTerminal, accessor: ServicesAccessor, instance?: ITerminalInstance, args?: unknown) => void | Promise<unknown> }
+): IDisposable {
+	const originalRun = options.run;
+	return registerTerminalAction({
+		...options,
+		run: (c, accessor, args) => {
+			const activeDetached = Iterable.find(c.service.detachedInstances, d => d.isFocused);
+			if (activeDetached) {
+				return originalRun(activeDetached as XtermTerminal, accessor, undefined, args);
```

Comment:
Yea, since it's an `IXtermTerminal` but some of the commands depend in internal methods that are not part of the public IXtermTerminal interface. We could also add them there, I was just following the existing pattern that depended on the  `xterm: XtermTerminal` on ITerminalInstance.

Reply:
Which methods are they? We are probably missing them on IXtermTerminal

> I was just following the existing pattern that depended on the xterm: XtermTerminal on ITerminalInstance.

I think this was an oversight and it should be using IXtermTerminal